### DISCUSSION TITLE
DebugAgentDxe and VMWare

### DISCUSSION THREAD

**Discussion by MiSimon** [tianocore/edk2#5307] [Q&A]
Hello,

i tried to get the DebugAgentDxe.efi from the [SourceLevelDebugPkg](https://github.com/tianocore/edk2/tree/master/SourceLevelDebugPkg) to work inside a VM running VMWare Workstation 16.

After i load the DebugAgentDxe.efi the VM freezes. Using the Intel UDK i can connect windbg to the Debug Agent, it seems that the VM gets stuck in a HLT instruction in a Module called YieldingDelayDxe, this module does not support unloading. If i set rip to the next instruction the VM continues to run but only for a short amount of time, then it gets stuck at the same position again.

Has anyone got the Debug Agent to work on VMWare and if so how?

Regards,
Michael

**Comment by sky5454** [comment]
VMware's  UEFI firmware  seems old, try  OVMF.
